include/aws/common/thread.h

```diff
@@ -237,6 +238,14 @@ AWS_COMMON_API void aws_thread_increment_unjoined_count(void);
  */
 AWS_COMMON_API void aws_thread_decrement_unjoined_count(void);
 
+/**
+ * Gets name of the current thread.
+ * Caller is responsible for destroying returned string.
+ * If thread does not have a name, NULL is returned.
+ * If call failed NULL is returned and aws error is raised.
+ */
+AWS_COMMON_API struct aws_string *aws_thread_current_name(struct aws_allocator *allocator);
```

Comment:
trivial: if you want to disambiguate between "has no data" and "error occurred", then change the signature to:
`int aws_thread_current_name(struct aws_allocator *, struct aws_string **out_name);`

that would be consistent with our existing APIs

to my knowledge, we have no APIs that clear aws_last_error() and then ask the user to check it themselves to see if something went wrong

Reply:
i've debated on this a little bit. i originally wrote it as int version, but i cant think of a case where user would care if its error or missing name. and passing out_name is a bit uglier to use. maybe we should just treat both cases the same and return NULL ptr.

I vaguely remember seeing this pattern in our code base (maybe in s3). But i do agree its ugly and probably should not be used here